### ISSUE TITLE
Error in Langflow AstraVectorise Component in Run JSON as Flow

### ISSUE THREAD

**Issue by edwinjosechittilappilly** [langflow-ai/langflow#2614]
Error in Langflow AstraVectorise Component in Run JSON as Flow

**Title**: Error in Langflow AstraVectorise Component in Run JSON as Flow

**Description**:
I am encountering an error when using the Langflow AstraVectorise component in the run JSON as Flow. The error message indicates an authentication issue with the AstraDB vector store.

**Error Message**:
```
Error performing search in AstraDBVectorStore: {"errors":[{"message":"Invalid vectorize authentication type: Credential error from SyncService : (GVC-005) Credential not found ","errorCode":"VECTORIZE_INVALID_AUTHENTICATION_TYPE"}]}
```

**Details**:
- The flow works correctly in the GUI of Langflow.
- The error occurs only when executing the run JSON as Flow.

**Environment**:
- Langflow version: `ragstack-ai-langflow ^1.0.6.post5`

**Steps to Reproduce**:
1. Create a flow in Langflow using the AstraVectorise component.
2. Export the flow as a run JSON.
3. Execute the run JSON as Flow.
4. Observe the authentication error message.

**Expected Behavior**:
The flow should execute without authentication errors, similar to how it operates within the Langflow GUI.

**Actual Behavior**:
An authentication error occurs, preventing the flow from executing properly.

**Additional Context**:
The issue seems to be related to the authentication mechanism used in the run JSON as Flow, which differs from the GUI execution.

**Screenshots**:
N/A

**Logs**:
```
Error performing search in AstraDBVectorStore: {"errors":[{"message":"Invalid vectorize authentication type: Credential error from SyncService : (GVC-005) Credential not found ","errorCode":"VECTORIZE_INVALID_AUTHENTICATION_TYPE"}]}
```

**Possible Solution**:
Review and correct the authentication process in the run JSON as Flow to match the one used in the GUI.
Also when Exported the TWEAKS the Authentication was empty dict.
```
"AstraVectorize-XlmUi": {
    "api_key_name": OpenAIKey,
    "authentication": {},
    "model_name": "text-embedding-3-small",
    "model_parameters": {},
    "provider": "OpenAI",
    "provider_api_key": "OpenAIKey"
  },
```

I tried replacing the  provider api keys with actual keys, but other compents like openAI LLM works with tweaks without explicitly mentioning the api keys in TWEAKS.

**Thank you for your assistance!**

**Comment by jordanrfrazier** [comment]
@edwinjosechittilappilly Looking at this, I see that you have `provider_api_key` set to "OpenAIKey". Just checking, is this the actual string you have, or just replacing your api key? This field should be the actual api key, and can be used when no API key is stored in Astra. This key will be manually placed in the header and passed along each request through the Data API. 
```
"AstraVectorize-XlmUi": {
    "api_key_name": OpenAIKey,
    "authentication": {},
    "model_name": "text-embedding-3-small",
    "model_parameters": {},
    "provider": "OpenAI",
    "provider_api_key": "OpenAIKey"
  },
```

The `api_key_name` should be set to the key name you provided to your astra collection on creation.

**Comment by edwinjosechittilappilly** [comment]
I did try both the approaches by providing the actual key as variable from ENV as provider_api_key, also tried to mention the api_key_name as same as the one stored in astraDB integration of OpenAI. but it resulted in same error.

**Comment by edwinjosechittilappilly** [comment]
Issue Solved from ragstack-ai-langflow = "^1.0.7.post2"
